### DISCUSSION TITLE
AudioWavServer is client connected

### DISCUSSION THREAD

**Discussion by OMENGT** [pschatzmann/arduino-audio-tools#961] [Q&A] [answered]
Hi
is there a way to tell if a client is connected to the AudioWAVServer from within the arduino loop? I would like to turn on an led when the is a client connected and turn it off when the is not a client connected

Many Thanks

**Comment by pschatzmann** [comment, marked as answer]
Why don't you just look at the source code ? 
It should not be too difficult to figure this out yourself.
One of the big advantages of Arduino is, that you have access to the full source code.

Hint if you look at the code you will find that the WiFiClient has a connected() method which is used in the logic of the AudioServer parent class as well.

Please let me know if you need further help...

**Reply by OMENGT** [reply]
thanks, I got it working
